File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/clinical/shared/social-history.model.ts
import {
    NgForm,
    FormGroup,
    FormControl,
    Validators,
    FormBuilder,
    ReactiveFormsModule
} from '@angular/forms';
import * as moment from 'moment/moment';
export class SocialHistory {
    public SocialHistoryId: number = 0;
    public PatientId: number = 0;
    public SmokingHistory: string = null;
    public AlcoholHistory: string = null;
    public DrugHistory: string = null;
    public Occupation: string = "";
    public FamilySupport: string = "";
    public CreatedDate: string = "";
    public Note: string = null;
    public CreatedBy: number = null;
    public ModifiedBy: number = null;
    public CreatedOn: string = null;
    public ModifiedOn: string = null;
    public SocialHistoryValidator: FormGroup = null;

    constructor() {
        var _formBuilder = new FormBuilder();
        this.SocialHistoryValidator = _formBuilder.group({
            //'Note': ['', Validators.compose([Validators.required,Validators.maxLength(200)])],
            //'Occupation': ['', Validators.compose([Validators.required, Validators.maxLength(50)])],
        });
    }

    public IsDirty(fieldName): boolean {
        if (fieldName == undefined)
            return this.SocialHistoryValidator.dirty;
        else
            return this.SocialHistoryValidator.controls[fieldName].dirty;
    }

    public IsValid():boolean{if(this.SocialHistoryValidator.valid){return true;}else{return false;}} public IsValidCheck(fieldName, validator): boolean {
        if (fieldName == undefined)
            return this.SocialHistoryValidator.valid;
        else
            return !(this.SocialHistoryValidator.hasError(validator, fieldName));
    }
    
}

File: Code/Components/DanpheEMR.ServerModel/ClinicalModels/SocialHistoryModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class SocialHistory
    {
        [Key]
        public int SocialHistoryId { get; set; }
        public int PatientId { get; set; }
        public string SmokingHistory { get; set; }
        public string AlcoholHistory { get; set; }
        public string DrugHistory { get; set; }
        public string Occupation { get; set; }
        public string FamilySupport { get; set; }
        public string Note { get; set; }
        public int? CreatedBy { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public virtual PatientModel Patient { get; set; }
    }
}
